src/NLog/Targets/ConsoleTarget.cs

```diff
@@ -127,6 +135,15 @@ public ConsoleTarget(string name) : this()
         protected override void InitializeTarget()
         {
             PauseLogging = false;
+            if (DetectConsoleAvailable)
+            {
+                PauseLogging = !IsConsoleAvailable();
+                if (PauseLogging && LoggingConfiguration != null)
+                {
+                    foreach (var loggingRule in LoggingConfiguration.LoggingRules)
+                        loggingRule.Targets.Remove(this);
```

Comment:
I'm not sure if this is the right approach. Would not expect it would change my config.